src/components/images/imageLoader.js

```diff
@@ -1,50 +1,90 @@
-define(['lazyLoader', 'imageFetcher', 'layoutManager', 'browser', 'appSettings', 'userSettings', 'require', 'css!./style'], function (lazyLoader, imageFetcher, layoutManager, browser, appSettings, userSettings, require) {
-    'use strict';
-
-    var self = {};
-
-    function fillImage(elem, source, enableEffects) {
+import lazyLoader from 'lazyLoader';
+import userSettings from 'userSettings';
+import 'css!./style';
+/* eslint-disable indent */
 
+    export function lazyImage(elem, source = elem.getAttribute('data-src')) {
         if (!elem) {
             throw new Error('elem cannot be null');
         }
 
         if (!source) {
-            source = elem.getAttribute('data-src');
+            return;
         }
 
-        if (!source) {
-            return;
+        fillImageElement(elem, source);
+    }
+
+    export function fillImage(entry) {
+        if (!entry) {
+            throw new Error('entry cannot be null');
         }
 
-        fillImageElement(elem, source, enableEffects);
+        var source = undefined;
+        if (entry.target) {
+            source = entry.target.getAttribute('data-src');
+        } else {
+            source = entry;
+        }
+
+        if (entry.intersectionRatio > 0 && source) {
+            fillImageElement(entry.target, source);
+        } else if (!source) {
+            emptyImageElement(entry.target);
+        }
     }
 
-    function fillImageElement(elem, source, enableEffects) {
-        imageFetcher.loadImage(elem, source).then(function () {
+    function fillImageElement(elem, url) {
+        if (url === undefined) {
+            throw new Error('url cannot be undefined');
+        }
+
+        let preloaderImg = new Image();
+        preloaderImg.src = url;
+
+        preloaderImg.addEventListener('load', () => {
+            if (elem.tagName !== "IMG") {
+                elem.style.backgroundImage = "url('" + url + "')";
+            } else {
+                elem.setAttribute("src", url);
+            }
 
-            if (enableEffects !== false) {
-                fadeIn(elem);
+            if (userSettings.enableFastFadein()) {
+                elem.classList.add('lazy-image-fadein-fast');
+            } else {
+                elem.classList.add('lazy-image-fadein');
             }
 
             elem.removeAttribute("data-src");
+            preloaderImg = null;
         });
     }
 
-    function fadeIn(elem) {
+    function emptyImageElement(elem) {
+        var url;
+
+        if (elem.tagName !== "IMG") {
+            url = elem.style.backgroundImage.slice(4, -1).replace(/"/g, "");
```

Comment:
what this magic does?

Reply:
It extracts the url from the value of background-image, which is like `url('something/somewhere.jpg')`.

Reply:
Regular expression for double-quotes. Maybe `/['"]/g`

Reply:
@MrTimscampi Could you fix single-quoted URL support here? Also are you sure there's no native way aside of regex-ing it away?

Reply:
There is no need to handle them since the JavaScript API will always return the browser-parsed version of the property, which will always use double quotes.